### PR TITLE
test : Adding announcement controller test

### PR DESCRIPTION
**Pull request by palakpatel311**

### CODE REVIEW

File: src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/ComplaintsTest.java [hunk 1]
```diff
@@ -31,38 +31,33 @@ public void setHeaders(String authorization) {
...
 
-    @When("^I send DELETE request to resource \"([^\"]*)\"$")
-    public void sendDeleteRequest(String resource) {
-        response = request.pathParams("complaintId", ComplaintsAsserts.postComplaintId).when().delete(resource);
-    }
+//    @When("^I send DELETE request to resource \"([^\"]*)\"$")
+//    public void sendDeleteRequest(String resource) {
+//        response = request.pathParams("complaintId", ComplaintsAsserts.postComplaintId).when().delete(resource);
+//    }
 }
```
Comment (ashwinimore) [inline src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/ComplaintsTest.java:61]: why comment and keep when a new step & step definition is added to replace this step?

File: src/test/resources/Features/API/announcementController.feature [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+Feature: Announcement-controller endpoints
+  Verify CRUD operations on announcement-controller endpoints
+  
+Background: Setting base URL
+    Given I set base URL
 
+  Scenario: Verify adding, updating and deleting a announcement
+    Given I set Headers with API key, "ROLE_ADMIN" authorization for announcements 
+    And I set "POST" announcements request body
+    When I send "POST" request to annoucement resource "rest/announcements"
```
Comment (ashwinimore) [inline src/test/resources/Features/API/announcementController.feature]: typo on announcement here and several places below

**Review by ashwinimore — APPROVED** [review]

File: src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/Base.java [hunk 3]
```diff
@@ -14,6 +16,8 @@
     private static String username;
     private static String password;
     private static final int HTTP_GOOD_RESPONSE = 200;
+    private static RequestSpecification request;
+    public static Response response;
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/Base.java]: This should probably be private, and have getter/setter methods

File: src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/AnnoucementsTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.When;
+import gov.healthit.chpl.aqa.api.asserts.AnnoucementsAsserts;
+import gov.healthit.chpl.aqa.api.payLoad.AnnouncementsPayload;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+
+public class AnnoucementsTest {
+
+    public static Response response;
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/api/stepDefinitions/AnnoucementsTest.java]: Should also be private

File: src/test/java/gov/healthit/chpl/aqa/api/asserts/AnnoucementsAsserts.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+import org.testng.Assert;
+import cucumber.api.java.en.Then;
+import gov.healthit.chpl.aqa.api.stepDefinitions.AnnoucementsTest;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+
+public class AnnoucementsAsserts {
+    private Response response;
+    public static int postAnnoucementId;
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/api/asserts/AnnoucementsAsserts.java]: Think this should be private as well

**Review by andlar — COMMENTED** [review]
Do all of these tests work as written? I know one thing that I saw as challenging was figuring out a way to pass information between steps in a Feature file; it looks like you're doing that here with a static class; is that generally accepted as the right way of doing that?

**Comment by palakpatel311** [comment]
@andlar Yes these test works. But i see confusion. I need to use same response/request and pass it through steps which became tough to do it this way. I could write feature file more consolidated way to avoid this. let me know if you have any other suggestion

**Review by andlar — APPROVED** [review]

**Pull request merged**